ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,295 @@
+
+    ctx.logger.info(
+        'Instance {0} removed from Load Balancer {1}.'
+        .format(instance_id, elb_name))
+    _remove_instance_from_elb_list_in_properties(instance_id)
+
+
+@operation
+def add_instance_to_elb(**_):
+
+    ctx.logger.info('Attempting to add_instance_to_elb.')
+
+    elb_name = \
+        utils.get_external_resource_id_or_raise(
+            'elb_name', ctx.target.instance)
+
+    instance_id = \
+        utils.get_external_resource_id_or_raise(
+            'instance_id', ctx.source.instance)
+
+    lb = _get_existing_elb(elb_name)
+
+    try:
+        lb.register_instances(instance_id)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+            raise NonRecoverableError('Instance not added to Load Balancer. '
+                                      '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Instance {0} added to Load Balancer {1}.'
+        .format(instance_id, elb_name))
+
+    _add_instance_to_elb_list_in_properties(instance_id)
+
+
+def _add_health_check_to_elb(elb, health_check):
+
+    hc = _create_health_check(health_check)
+
+    try:
+        elb.configure_health_check(hc)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Health check not added to Load Balancer. '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Health check added to Load Balancer {0}.'
+        .format(elb.name))
+
+
+def use_external_elb(**_):
+
+    utils.use_external_resource(ctx.node.properties)
+
+
+@operation
+def delete_elb(**_):
+
+    if ctx.node.properties['use_external_resource']:
+        utils.unassign_runtime_property_from_resource(
+            constants.EXTERNAL_RESOURCE_ID, ctx.instance)
+        return
+
+    ctx.logger.info('Attempting to delete Load Balancer.')
+
+    elb_name = ctx.node.properties['elb_name']
+
+    lb = _get_existing_elb(elb_name)
+
+    try:
+        lb.delete()
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Load Balancer {0} not deleted. '
+                                  .format(str(e)))
+
+    ctx.logger.info(
+        'Load Balancer {0} deleted. '
+        .format(elb_name))
+    if 'elb_name' in ctx.instance.runtime_properties:
+        ctx.instance.runtime_properties.pop('elb_name')
+
+    return
+
+
+def _create_health_check(user_health_check):
+
+    health_check = {
+        'interval': constants.HEALTH_CHECK_INTERVAL,
+        'healthy_threshold': constants.HEALTH_CHECK_HEALTHY_THRESHOLD,
+        'timeout': constants.HEALTH_CHECK_TIMEOUT,
+        'unhealthy_threshold': constants.HEALTH_CHECK_UNHEALTHY_THRESHOLD
+    }
+
+    health_check.update(user_health_check)
```

Comment:
better solution: Since healthcheck is a local operation and does that verification for me, will put it in a try block which will throw a NonRecoverableError if it fails, since it can only fail on bad parameters.